### PR TITLE
Create full extracts as needed via the UI or at command-line

### PR DESCRIPTION
**Pull request by lwrubel**
To set up:
- add a full_datasets directory to tweetset_data as described in the README
- in your `docker-compose.yml`, add the volume for the full_dataset into the containers for server, server-flaskrun, and worker, as in `example.cluster-primary.docker-compose.yml`

Testing areas:
- Create an extract on the command-line and confirm a directory is created in full_datasets with the CSV, JSON, and id files. 
- In the UI, select a dataset and then do not add any parameters to the parameters form. Confirm that the files get created in the dataset exports tab, if they did not already exist. If they did already exist, confirm that you see the appropriate files for your IP range. (Change the IP_RANGE definition in .env to simulate a non-GW user)
- Confirm that a custom extract, not the full extract, is created if parameters are used. 
- Confirm that the content on the preview tab is accurate when selecting a full dataset (both by clicking the `Preview` button and by using the tab.

### CODE REVIEW

**Comment by kerchner** [comment]
@dolsysmith and/or @lwrubel can you share your docker-compose.yml with me?   Although I see that, inside my container, `tweetsets_server.py` is the correct version for this branch, I am getting an `Error: No such command 'create-extract'` error.

File: docker/loader.docker-compose.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 services:
   loader:
-      image: gwul/tweetsets-loader
-      #build:
-      #    context: ..
-      #    dockerfile: Dockerfile-loader
+      #image: gwul/tweetsets-loader
+      build:
+          context: ..
+          dockerfile: Dockerfile-loader
```
Comment (kerchner) [inline docker/loader.docker-compose.yml]: @dolsysmith @lwrubel Should this change be part of the PR, or was it intended for development/testing only?

File: tweetset_server.py [hunk 2]
```diff
@@ -20,6 +21,7 @@
 app = Flask(__name__)
 app.config['SECRET_KEY'] = os.environ.get('SECRET_KEY', 'secret')
 app.config['DATASETS_PATH'] = '/tweetsets_data/datasets'
+app.config['FULL_DATASETS_PATH'] = '/tweetsets_data/full_datasets'
```
Comment (kerchner) [inline tweetset_server.py:24]: @dolsysmith @lwrubel Should this path be hard-coded, or should it take the path from the environment?

**Review by kerchner — COMMENTED** [review]
Please see two comments in the code.  Everything functions perfectly, although I understand there is a problem sending the email to the admin upon completion of `create-extract` (is it mainly a problem with the `sfm_no_reply@gwu.edu` email account?  I have not tried it yet with my own account.).  In the interest of time, the admin email notification can certainly be carved off as a separate bugfix ticket, while allowing the rest of the PR to proceed.

I should also note that I got this working with Laura's `docker-compose.yml` and `.env`, and I did not have a chance to get everything working with the example docker-compose files in this branch (despite trying).  Please let me know if you feel this is a risk and I should go back and do this as well.

**Comment by kerchner** [comment]
Trying to send the admin notification email from my own email account resulted in this:
image

File: tweetset_server.py (hunk 2 above)
Comment (dolsysmith) [reply tweetset_server.py:24]: The path in the `.ENV` is mapped to the Docker volume, and this line refers to the Docker volume path. I was keeping it parallel with the line above.

File: docker/loader.docker-compose.yml (hunk 1 above)
Comment (dolsysmith) [reply docker/loader.docker-compose.yml]: You're right -- this should not be part of the PR. I can change it back before merging.

File: tweetset_server.py (hunk 2 above)
Comment (kerchner) [reply tweetset_server.py:24]: Got it, thanks

**Comment by dolsysmith** [comment]
@kerchner Did you "enable less secure apps" in your Gmail account? It's under Account Settings -- Security. It's necessary for the SMTP connection that TweetSets relies on.

**Comment by kerchner** [comment]
@dolsysmith Thanks, I've updated the setting and am retrying `create-extract`

**Comment by kerchner** [comment]
Still not getting the email.

**Comment by kerchner** [comment]
I see in my sent email folder that the app sent the email (to me) on my behalf.  So that seems to work.

**Comment by kerchner** [comment]
Please proceed with merging.

**Pull request merged**